code/modules/urist/vehicles/motorcycle.dm

```diff
@@ -49,4 +49,251 @@
+	..(user)
+	switch(buildstate)
+		if(1) user << "It has a loose wheel well in place."
+		if(2) user << "It has a wheel mount welded in place."
+		if(3) user << "It has a back tire loosely attached."
+		if(4) user << "It has both tires loosely attached."
+		if(5) user << "It has both tires firmly attached."
+		if(6) user << "It has both tires firmly attached and a transmission loosely in place."
+		if(7) user << "It has both tires firmly attached and a transmission firmly in place."
+		if(8) user << "It has both tires, a transmission and a loosely attached battery."
+		if(9) user << "It has both tires, a transmission and a firmly attached battery."
+
+/obj/structure/vehicle_frame/motorcycle/attackby(obj/item/W as obj, mob/user as mob)
+	if(istype(W,/obj/item/stack/rods))
+		if(buildstate == 0)
+			var/obj/item/stack/rods/R = W
+			if(R.use(3))
+				user << "<span class='notice'>You assemble a backbone of rods, constructing a crude wheel well.</span>"
+				buildstate++
+				update_icon()
+			else
+				user << "<span class='notice'>You need at least three rods to complete this task.</span>"
+			return
+
+	else if(istype(W,/obj/item/weapon/weldingtool))
+		if(buildstate == 1)
+			var/obj/item/weapon/weldingtool/T = W
+			if(T.remove_fuel(0,user))
+				if(!src || !T.isOn()) return
+				playsound(src.loc, 'sound/items/Welder2.ogg', 100, 1)
+				user << "<span class='notice'>You weld the rods into place.</span>"
+			buildstate++
+		return
+
+	else if(istype(W,/obj/item/vehicle_part/tire))
+		if(buildstate == 2)
+			user << "<span class='notice'>You slide a tire into the back wheel mount.</span>"
+			buildstate++
+			update_icon()
+			return
+
+		else if(buildstate == 3)
+			user << "<span class='notice'>You slide a tire into the front wheel mount.</span>"
+			buildstate++
+			update_icon()
+			return
+
+	else if(istype(W,/obj/item/weapon/wrench))
+		if(buildstate == 4)
+			user << "<span class='notice'>You secure the tires into the motorcycle frame.</span>"
+			buildstate++
+			return
+
+		else if(buildstate == 8)
+			user << "<span class='notice'>You secure the battery into the frame.</span>"
+			new /obj/vehicle/bike/motorcycle(get_turf(src))
+			qdel(src)
+			return
+
+	else if(istype(W,/obj/item/vehicle_part/transmission))
+		if(buildstate == 5)
+			user << "<span class='notice'>You slide the transmission into the frame.</span>"
+			buildstate++
+			update_icon()
+			return
+
+	else if(istype(W,/obj/item/weapon/screwdriver))
+		if(buildstate == 6)
+			user << "<span class='notice'>You secure the transmission into the frame.</span>"
+			buildstate++
+			return
+
+	else if(istype(W,/obj/item/vehicle_part/battery))
+		if(buildstate == 7)
+			user << "<span class='notice'>You slide the battery into the frame.</span>"
+			buildstate++
+			update_icon()
+			return
+
+	else
+		..()
+
+/obj/structure/vehicle_frame/motorcycle/update_icon()
+	icon_state = "bike_frame[buildstate]"
+
+/obj/vehicle/bike/motorcycle
```

Comment:
Might be worthwhile to make a prototype for all persistent-movement type vehicles, considering we've got all the cars and stuff like that.